src/averageAges.js

```diff
@@ -55,10 +76,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const peopleWhitMother = people.filter(person => !onlyWithSon
+    ? people.find(women => person.mother === women.name)
+    : (people.find(women => person.mother === women.name
+      && person.sex === MALE)
+    ));
```

Comment:
To simplify this code, you can first filter all `people` separately and leave only those who have mothers, and then check if `onlyWithSon` equal to trues, then filter that those who remain are only men.